src/daft-core/src/series/array_impl/binary_ops.rs

```diff
@@ -232,23 +232,85 @@ impl SeriesBinaryOps for ArrayWrapper<StructArray> {}
 impl SeriesBinaryOps for ArrayWrapper<MapArray> {}
 impl SeriesBinaryOps for ArrayWrapper<ExtensionArray> {}
 impl SeriesBinaryOps for ArrayWrapper<Decimal128Array> {}
-impl SeriesBinaryOps for ArrayWrapper<DateArray> {}
+impl SeriesBinaryOps for ArrayWrapper<DateArray> {
+    fn add(&self, rhs: &Series) -> DaftResult<Series> {
+        use DataType::*;
+        let output_type = (self.data_type() + rhs.data_type())?;
+        let lhs = self.0.clone().into_series();
+        match rhs.data_type() {
+            Duration(tu) => {
+                let intermediate_timestamp_type = Timestamp(*tu, None);
+                let lhs = lhs.cast(&intermediate_timestamp_type)?.as_physical()?;
+                let rhs = rhs.as_physical()?;
+                let physical_result = lhs.add(rhs)?.cast(&intermediate_timestamp_type)?;
+                physical_result.cast(&output_type)
```

Comment:
For arithmetic between date/duration, the precision will never match because date is measured in days, whereas our timeunits for timestamp / duration are measured in nanoseconds up to seconds. 

To overcome this, I leverage the existing functionality to cast date to timestamp, which will introduce the same level of precision as the duration.

Reply:
Hmm yeah nice catch. I don't think we need a dedicated `days` timeunit though, which will introduce a bunch of extra work given that arrow2 does not have `days` timeunit. I added a function instead to just cast a duration array to an i32 array in the days unit via division.